app/services/RepaymentService.scala

```diff
@@ -17,27 +17,39 @@
 package services
 
 import connectors.RepaymentConnector
-import models.core.RepaymentJourneyResponseModel
+import models.core.RepaymentJourneyResponseModel.{RepaymentJourneyErrorResponse, RepaymentJourneyModel}
 import play.api.Logger
 import uk.gov.hmrc.http.HeaderCarrier
 
-import javax.inject.Singleton
-import javax.inject.Inject
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
-class RepaymentService @Inject()(val repaymentConnector: RepaymentConnector){
+class RepaymentService @Inject()(val repaymentConnector: RepaymentConnector) {
 
   def start(nino: String, fullAmount: BigDecimal)
-           (implicit headerCarrier: HeaderCarrier): Future[RepaymentJourneyResponseModel] = {
+           (implicit headerCarrier: HeaderCarrier): Future[Either[Throwable, String]] = {
```

Comment:
you might pass fullAmount as Option[BigDecimal] here and do extract absolute value, for None value just raise an exception

Reply:
done